package.json

```diff
@@ -3,7 +3,9 @@
   "version": "1.0.0",
   "main": "index.js",
   "scripts": {
-    "test": "mdspell --en-gb -ranx 'content/**/*.md'"
+    "spellcheck": "mdspell --en-gb -ranx 'content/**/*.md'",
+    "broken-link": "blcl -ro ./public/",
```

Comment:
This requires that the site be built, yet there's nothing here which checks that the build is up to date.

I'm also seeing it hitting _external_ urls, which seems likely to generate false failures on CI if the network is down. Worryingly some of the "external" urls it hits are our own site!
```
Getting links from: http://localhost:39037/tutorials/kit-assembly/
├───OK─── https://github.com/sourcebots/docs/tree/master/content/tutorials/kit-assembly.md
├───OK─── http://localhost:39037/tutorials/kit-assembly.files/camcons.png
├───OK─── https://docs.sourcebots.co.uk/tutorials/kit-assembly.files/camcons.png
├───OK─── https://docs.sourcebots.co.uk/tutorials/kit-assembly.files/motor-board.png
├───OK─── https://docs.sourcebots.co.uk/tutorials/kit-assembly.files/motor_and_camcon.png
├───OK─── https://docs.sourcebots.co.uk/tutorials/kit-assembly.files/pi-power-cable.png
├───OK─── https://docs.sourcebots.co.uk/tutorials/kit-assembly.files/pi.png
├───OK─── https://docs.sourcebots.co.uk/tutorials/kit-assembly.files/power-board.png
├───OK─── https://docs.sourcebots.co.uk/tutorials/kit-assembly.files/servo-assembly.png
Finished! 84 links found. 75 excluded. 0 broken.
```

Reply:
I believe this is because there are places which the main URL is baked into the site. I've been planning to go through and remove these at some point. Checking external URLs isnt great, but it's a necessary test, and we can hopefully well assume that CircleCI has a good network, and the sites we depend on are fairly dependable (if Pythons docs go down, it's a bigger issue than just 'our site doesnt build')

Reply:
I'd agree with Jake here - I'd prefer the tests failing if CircleCI has network problems (which we can manually check and ignore) to letting broken links slip past

Reply:
However, if Python's docs go down we _don't_ want that to block our ability to update our own docs.

Hitting the network in tests also means that you can't reliably run the tests reproducibly, nor run them when developing offline. Even when online it means that you're forced to hit all those websites just to run our tests.

I'd _much_ rather have a way to check local-only, even if there's a separate command which then runs a fuller test that includes the remote urls.